tutorials/08-implementing-a-custom-feature.md

```diff
@@ -0,0 +1,274 @@
+the \ref ignition::physics::Feature "Feature" is implemented. These are the most
+common "proxy objects" that are inherited from `Entity` class:
+  - \ref ignition::physics::Feature::Engine "Engine": Placeholder class for the
+    Engine API. This class serves metadata for the physics engine (for example
+    the \ref ignition::physics::GetEngineInfo "GetEngineInfo" feature).
+    Every Engine feature **must** inherit this class.
+  - \ref ignition::physics::Feature::Joint "Joint": defines physics concept `Joint` behaviors (for example
+    the \ref ignition::physics::GetBasicJointState "GetBasicJointState"
+    feature).
+  - \ref ignition::physics::Feature::Link "Link": defines physics concept `Link` structure.
+  - \ref ignition::physics::Feature::Model "Model": defines physics concept `Model` structure (for example
+    the \ref ignition::physics::GetLinkFromModel "GetLinkFromModel"
+    feature including both `Link` and `Model` objects).
+  - \ref ignition::physics::Feature::Shape "Shape": defines physics concept `Shape` structure (for example
+    the \ref ignition::physics::GetShapeKinematicProperties "GetShapeKinematicProperties"
+    feature).
+  - \ref ignition::physics::Feature::World "World": defines physics concept `Shape` structure (for example
+    the \ref ignition::physics::dartsim::RetrieveWorld "RetrieveWorld" feature
+    in `dartsim` plugin).
+
+  Note that these object classes are not mutually exclusive and could be defined in conjunction together to describe the `Feature`. There are also other uncommon
+  objects defined depending on feature functionality, for example, the \ref ignition::physics::SetFreeGroupWorldPose::FreeGroup "FreeGroup"
+  object in `SetFreeGroupWorldPose` feature. For more information about the
+  physics concepts, please refer to
+  \ref physicsconcepts "Ignition Physics simulation concepts" tutorial.
+- \ref ignition::physics::Feature::Implementation "Implementation" interfaces
+the actual physics engines API for the custom feature. It has
+\ref ignition::physics::Feature::Implementation::InitiateEngine "InitiateEngine"
+ to trigger physics engine initiation to provide the required functionalities.
+
+Moreover, we can define dependencies between custom `Features`:
+- By default, a blank feature will not require any other features.
+If the custom feature does require some other set of features,
+then it should be inherited from
+\ref ignition::physics::FeatureWithRequirements "FeatureWithRequirements" class,
+and provided a list of the `Features` required.
+- By default, a blank feature will not conflict with any other features. If
+the custom feature does conflict with some other set of features, then it should
+be inherited from
+\ref ignition::physics::FeatureWithConflicts "FeatureWithConflicts" class,
+and provided a list of the conflicting `Features`. The conflicting `Features`
+will not run at the same time when requested.
+
+### Define the custom feature
+
```

Comment:
It'd be helpful to establish a folder structure for user to follow along. So that they know what code block to add to which file throughout . Then in the end, they will have a working example folder which they can compile locally.

Reply:
@anindex friendly ping on this one :)

Reply:
I did add a folder structure a few line below. Im not sure if that what you mean :)

Reply:
Okay I added the folder structure for this section :)